src/components/ItemModal.ts

```diff
@@ -111,39 +111,42 @@ export class ItemModal extends Modal {
             .setDynamicTooltip()
         );
     }
-
-    new Setting(contentEl)
-    .setName("Do Date")
-    .addMomentFormat((cb) =>
-        cb
-        .setDefaultFormat(this.dateFormat)
-        .setValue(this.successPlanItem.do_date != "" ? this.successPlanItem.do_date.format(this.dateFormat) : "")
-        .onChange(async (val) => {
-            this.successPlanItem.do_date = moment(val);
-        })
-    );
-
-    new Setting(contentEl)
-    .setName("Due Date")
-    .addMomentFormat((cb) =>
-        cb
-        .setDefaultFormat(this.dateFormat)
-        .setValue(this.successPlanItem.due_date != "" ? this.successPlanItem.due_date.format(this.dateFormat) : "")
-        .onChange(async (val) => {
-            this.successPlanItem.due_date = moment(val);
-        })
-    );
-
-    new Setting(contentEl)
-    .setName("Closing Date")
-    .addMomentFormat((cb) =>
-        cb
-        .setDefaultFormat(this.dateFormat)
-        .setValue(this.successPlanItem.closing_date != "" ? this.successPlanItem.closing_date.format(this.dateFormat) : "")
-        .onChange(async (val) => {
-            this.successPlanItem.closing_date = moment(val);
-        })
-    );
+    
+    const doDate_setting_item = contentEl.createEl("div", {cls: "setting-item"});
+    const doDate_setting_item_info = contentEl.createEl("div", {cls: "setting-item-info"});
+    doDate_setting_item_info.append(contentEl.createEl("div", { text: 'Do Date', cls: "setting-item-name"}));
+    const doDate_setting_item_control = contentEl.createEl("div", {cls: "setting-item-control"});
+    const doDate_picker = contentEl.createEl("input", {type: "text", value: this.successPlanItem.do_date != "" ? this.successPlanItem.do_date._i : ""  });
+    doDate_picker.onfocus = () => {doDate_picker.type = 'date'; doDate_picker.className="dropdown"; doDate_picker.value = this.successPlanItem.do_date.format("YYYY-MM-DD");};
+    doDate_picker.onchange = async (event) => {this.successPlanItem.do_date = moment(event.target.value)};
+    doDate_picker.onblur = () => {doDate_picker.type = 'text'; doDate_picker.value = this.successPlanItem.do_date.format(this.dateFormat)};
+    doDate_setting_item_control.append(doDate_picker);
+    doDate_setting_item.append(doDate_setting_item_info);
+    doDate_setting_item.append(doDate_setting_item_control);
```

Comment:
In order to follow the DRY (Don't Repeat Yourself) philosophy, please refactor this code into a single method so that it can be used to set up each date setting.